gaphor/diagram/general/generalpropertypages.py

```diff
@@ -88,3 +97,74 @@ def construct(self):
     def _on_field_change(self, entry, field_name):
         text = entry.get_text()
         setattr(self.item, field_name, text)
+
+
+FILTER_IMAGES = [
+    ("Images", "*.png, *.jpg, *.jpeg", "image/*"),
+]
+
+
+@PropertyPages.register(PictureItem)
+class PicturePropertyPage(PropertyPageBase):
+    """Edit picture settings"""
+
+    def __init__(self, subject):
+        self.subject = subject
+        self.watcher = subject and subject.watcher()
+
+    def construct(self):
+        subject = self.subject
+
+        if not subject:
+            return
+
+        builder = new_builder(
+            "picture-editor",
+            signals={
+                "select-picture": (self._on_select_picture_clicked,),
+                "set-default-size": (self._on_default_size_clicked),
+            },
+        )
+        return builder.get_object("picture-editor")
+
+    @transactional
+    def _on_select_picture_clicked(self, button):
+        open_file_dialog(
+            gettext("Select a picture..."), self.open_files, filters=FILTER_IMAGES
+        )
+
+    @transactional
+    def open_files(self, filenames):
```

Comment:
I really like the resize that happens when we run the set Default Size button. Could we run this resize when we first load an image as well?

Reply:
Yeah, we are doing this as we discussed yesterday.